src/electron/ipc/ipc-event-action.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+export class IpcEventAction {
```

Comment:
I think this approach seems reasonable, but I think it would make more sense to call this an `AsyncAction`; I don't think what it's actually doing is really related to `IpcEvent`s except insofar as that happens to be how it's consumed